client/dashboard/task-list/tasks/payments/paypal.js

```diff
@@ -0,0 +1,257 @@
+import interpolateComponents from 'interpolate-components';
+
+/**
+ * WooCommerce dependencies
+ */
+import { WC_ADMIN_NAMESPACE } from 'wc-api/constants';
+import { Form, Link } from '@woocommerce/components';
+import withSelect from 'wc-api/with-select';
+
+class PayPal extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			connectURL: '',
+			showManualConfiguration: props.manualConfig,
+		};
+
+		this.updateSettings = this.updateSettings.bind( this );
+	}
+
+	componentDidMount() {
+		const { showManualConfiguration } = this.state;
+
+		const query = getQuery();
+		// Handle redirect back from PayPal
+		if ( query[ 'paypal-connect' ] ) {
+			if ( '1' === query[ 'paypal-connect' ] ) {
+				this.props.markConfigured( 'paypal' );
+				this.props.createNotice(
+					'success',
+					__( 'PayPal connected successfully.', 'woocommerce-admin' )
+				);
+				return;
+			}
+
+			/* eslint-disable react/no-did-mount-set-state */
+			this.setState( {
+				showManualConfiguration: true,
+			} );
+			/* eslint-enable react/no-did-mount-set-state */
+			return;
+		}
+
+		if ( ! showManualConfiguration ) {
+			this.fetchOAuthConnectURL();
+		}
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		if (
+			true === prevState.showManualConfiguration &&
+			false === this.state.showManualConfiguration
+		) {
+			this.fetchOAuthConnectURL();
+		}
+
+		if ( false === prevProps.optionsIsRequesting && true === this.props.optionsIsRequesting ) {
+			this.props.setRequestPending( true );
+		}
+
+		if ( true === prevProps.optionsIsRequesting && false === this.props.optionsIsRequesting ) {
+			this.props.setRequestPending( false );
+		}
+	}
+
+	async fetchOAuthConnectURL() {
+		this.props.setRequestPending( true );
+		try {
+			const result = await apiFetch( {
+				path: WC_ADMIN_NAMESPACE + '/onboarding/plugins/connect-paypal',
+				method: 'POST',
+			} );
+			if ( ! result || ! result.connectUrl ) {
+				this.props.setRequestPending( false );
+				this.setState( {
+					showManualConfiguration: true,
+				} );
+				return;
+			}
+			this.props.setRequestPending( false );
+			this.setState( {
+				connectURL: result.connectUrl,
+			} );
+		} catch ( error ) {
+			this.props.setRequestPending( false );
+			this.setState( {
+				showManualConfiguration: true,
+			} );
+		}
+	}
+
+	renderConnectButton() {
+		const { connectURL } = this.state;
+		return (
+			<Button isPrimary isDefault href={ connectURL }>
+				{ __( 'Connect', 'woocommerce-admin' ) }
+			</Button>
+		);
+	}
+
+	async updateSettings( values ) {
+		const { createNotice, isSettingsError, updateOptions, markConfigured } = this.props;
+
+		this.props.setRequestPending( true );
+		await updateOptions( {
+			woocommerce_ppec_paypal_settings: {
+				...this.props.options.woocommerce_ppec_paypal_settings,
+				api_username: values.api_username,
+				api_password: values.api_password,
+				enabled: 'yes',
+			},
+		} );
+
+		if ( ! isSettingsError ) {
```

Comment:
Should we add a success message here too when settings are updated successfully?  After saving PayPal settings manually I get bumped back to the task list without any other notice; may be a little confusing to some users.

Reply:
Good catch, thanks! Added a message on manual save for both PayPal and Stripe: https://github.com/woocommerce/woocommerce-admin/pull/2897/commits/19f561e2b1dcd232e2be1357d785f19ae38b3ed0